src/harberger-lib/components/BuyInput.res

```diff
@@ -49,12 +48,10 @@ let make = (
     None
   }, (deposit, setDepositSlider))
 
-  let (debouncedSetDeposit, _, _) = useDebouncedCallback(.
-    // debounced function
+  let {callback: debouncedSetDeposit, _} = useDebouncedCallback(.
     setDeposit,
-    // delay in ms
     50,
-    {maxWait: Some(500), leading: None, trailing: None},
+    {maxWait: Some(500), leading: Some(true), trailing: Some(true)},
```

Comment:
leading = it changes immediately (makes it feel responsive)

trailing = it always changes at the end (always updates to the final value)